src/NLog/Internal/MultiFileWatcher.cs

```diff
@@ -107,33 +136,33 @@ internal void Watch(string fileName)
                 return;
             }
 
-            var watcher = new FileSystemWatcher
+            lock (this)
```

Comment:
Can we remove the `lock` from `this` (I know it was like this before, but I think it would be an improvement if it's possible)